slick-testkit/src/main/scala/com/typesafe/slick/testkit/util/StandardTestDBs.scala

```diff
@@ -292,6 +295,9 @@ object DerbyDB {
 abstract class HsqlDB(confName: String) extends InternalJdbcTestDB(confName) {
   val profile = HsqldbProfile
   val jdbcDriver = "org.hsqldb.jdbcDriver"
+  // Hsqldb has valid "select for update" syntax, but in testing, it either takes a whole table lock or no exclusive
+  // lock at all, so exclude from ForUpdate testing
+  override def capabilities = super.capabilities - TestDB.capabilities.selectForUpdateRowLocking
```

Comment:
If you think it *should* be OK for Hsqldb but get runtime failures, it's probably just a bug or limitation in Hsqldb. In that case I'd make the driver advertise the capability (as you did) and add a note to the scaladoc that there are limitations. Excluding the test with an extra capability is probably the best option. I don't have a better idea, either.